hpgeom/healpix_geom.c

```diff
@@ -1325,6 +1325,13 @@ void query_ellipse(healpix_info *hpx, double ptg_theta, double ptg_phi, double s
         return;
     }
 
+    if ((semi_major <= 0) || (semi_minor <= 0) || (semi_major < semi_minor)) {
+        snprintf(err, ERR_SIZE, "query_ellipse must have semi_major and semi_minor positive "
+                 "and semi_major >= semi_minor.");
+        *status = 0;
+        return;
+    }
+
```

Comment:
Does this block raise a python error too? Maybe the status does that? In general, error statement printouts at the C level are annoying for python users because they cannot suppress them or catch them. It'd be better to raise a python error directly.

Reply:
First of all, this is a belt-and-suspenders approach (there is a check at the python/c interface) but in case somebody uses this code standalone it's nice to have an additional check to keep things from going crazy.
Second of all, when the status is set then the error string is put into a python exception and raised there: https://github.com/LSSTDESC/hpgeom/blob/7b6effe58a1d0064e54a8045a25a0d4d3a363788/hpgeom/hpgeom.c#L727-L730. (and notice this is a `snprintf` to a string and not a `[f]printf` to an output device).

Or ... maybe the confusion is ... this code is pure c, no python at all.  The python interface is entirely contained in `hpgeom.c`.